internal/xds/translator/listener.go

```diff
@@ -92,6 +93,21 @@ func addXdsHTTPFilterChain(xdsListener *listener.Listener, irListener *ir.HTTPLi
 		}},
 	}
 
+	// Enable grpc-web filter for HTTP2
+	if irListener.IsHTTP2 {
```

Comment:
always enables this for a http2 listener?
what about use an annotation to enable or close this

Reply:
hoping this project stays away from annotations and env variables due to the lack to discoverability and validation

reg auto enable of grpc-web for HTTP2 listener, we have two options
1. wait for upstream gateway-api to dictate whether grpc-web can be implicitly assumed for GRPCRoute or add an opt-in knob in the API. This process is bound to take a much longer time since its a common API across multiple implementations.
2. we auto enable it for GRPCRoute based listeners (which will have `IsHTTP2` set), if users want to only select pure grpc, they could do so by matching on `Content-Type: application/grpc`

I prefer 2. over 1.